mergify_engine/context.py

```diff
@@ -82,6 +83,63 @@ def get_repository(self, name: github_types.GitHubRepositoryName) -> "Repository
             self.repositories[name] = repository
         return self.repositories[name]
 
+    TEAM_MEMBERS_CACHE_KEY_PREFIX = "team_members"
+    TEAM_MEMBERS_CACHE_KEY_DELIMITER = "/"
+    TEAM_MEMBERS_EXPIRATION = 3600  # 1 hour
+
+    @classmethod
+    def _team_members_cache_key_for_repo(
+        cls,
+        owner_id: github_types.GitHubAccountIdType,
+    ) -> str:
+        return (
+            f"{cls.TEAM_MEMBERS_CACHE_KEY_PREFIX}"
+            f"{cls.TEAM_MEMBERS_CACHE_KEY_DELIMITER}{owner_id}"
+        )
+
+    @classmethod
+    async def clear_team_members_cache_for_team(
+        cls,
+        redis: utils.RedisCache,
+        owner: github_types.GitHubAccount,
+        team_slug: github_types.GitHubTeamSlug,
+    ) -> None:
+        await redis.hdel(
+            cls._team_members_cache_key_for_repo(owner["id"]),
+            team_slug,
+        )
+
+    @classmethod
+    async def clear_team_members_cache_for_org(
+        cls, redis: utils.RedisCache, user: github_types.GitHubAccount
+    ) -> None:
+        await redis.delete(cls._team_members_cache_key_for_repo(user["id"]))
+
+    async def get_team_members(
+        self, team_slug: github_types.GitHubTeamSlug
+    ) -> typing.List[github_types.GitHubLogin]:
+
+        key = self._team_members_cache_key_for_repo(self.owner_id)
+        members_raw = await self.redis.hget(key, team_slug)
+        if members_raw is None:
+            members = [
+                github_types.GitHubLogin(member["login"])
+                async for member in self.client.items(
+                    f"/orgs/{self.owner_login}/teams/{team_slug}/members"
+                )
+            ]
+            # TODO(sileht): move to msgpack when we remove redis-cache connection
+            # from decode_responses=True (eg: MRGFY-285)
+            pipe = await self.redis.pipeline()
+            await pipe.hset(key, team_slug, json.dumps(members))
```

Comment:
do you have to reencode here since you could get GitHub response directly?

Reply:
What do you mean by reencode ?   if  you mean str.encode() the json dumps, it's not needed because the redis client is configured to do it.
also members is not a GitHub response but just list of GitHubLogin (str).

Reply:
I meant that GitHub replies with a JSON that we could store right away to Redis.
But I see your points, we're only storing logins.